Marlin/Marlin_main.cpp

```diff
@@ -7459,8 +7459,10 @@ void prepare_move() {
           r_X = -offset[X_AXIS],  // Radius vector from center to current location
           r_Y = -offset[Y_AXIS],
           rt_X = target[X_AXIS] - center_X,
-          rt_Y = target[Y_AXIS] - center_Y;
+          rt_Y = target[Y_AXIS] - center_Y,
+          final_dest[NUM_AXIS];
 
+    memcpy(final_dest, target, sizeof(final_dest));
     // CCW angle of rotation between position and target from the circle center. Only one atan2() trig computation required.
```

Comment:
We get `target` as a reference to destination. Without saving we don't end at the right point.